mockgcp/mockserviceusage/serviceusagev1.go

```diff
@@ -273,6 +275,47 @@ func (s *ServiceUsageV1) ListServices(ctx context.Context, req *pb.ListServicesR
 		})
 	}
 
+	// Implement field filtering
+	// TODO: Can we mame this generic?  Is it used in other APIs?
+	{
+		md, _ := metadata.FromIncomingContext(ctx)
+		query := ""
+		if md != nil {
+			for _, v := range md.Get("query") {
+				query = v
+			}
+		}
+		fields := ""
+		if query != "" {
+			q, err := url.ParseQuery(query)
+			if err != nil {
+				return nil, status.Errorf(codes.InvalidArgument, "unable to parse query %q: %v", query, err)
+			}
+			fields = q.Get("fields")
+		}
+		if fields != "" {
+			include := make(map[string]bool)
+			for _, field := range strings.Split(fields, ",") {
+				include[field] = true
```

Comment:
This implementation only supports a flat list of fields. Real field masks support nesting, e.g., `services(name,parent)`. A simple `strings.Split` won't handle this. While maybe not required for the immediate need, it makes the mock less realistic.